### PR TITLE
Responsive => TypeScript

### LINKED ISSUE #189: Convert all Responsive subcomponents to TypeScript

### PR DESCRIPTION
**Pull request by mdespuits**
I tried to do this a bit more piecemeal, but it seems like there are too many inter-dependencies for that to happen cleanly. Went ahead and converted everything related to Responsive-ness to TS.

Fixes #189

### CODE REVIEW

File: src/components/Responsive/Element/Element.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactElement } from 'react';
+import Context from '../Context';
+
+const Element: React.FC = ({ children, ...passedProps }) => (
```
Comment (pixelbandito) [inline src/components/Responsive/Element/Element.tsx:4]: Did it complain when typing children with something like `children?: React.ReactNode` ?

File: src/components/Responsive/Grid/Grid.tsx [hunk 2]
```diff
@@ -1,10 +1,34 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ReactElement } from 'react';
 
 import UnresponsiveGrid from '../../Grid';
 import Context from '../Context';
 
-const Grid = ({ breakpoints, children, columns, gutter, ...passedProps }) => {
+type GridGutter = number | string;
```
Comment (pixelbandito) [inline src/components/Responsive/Grid/Grid.tsx:6]: Super minor, but I'd usually put an empty newline before the multiline `type GridDefinition ...`

Not a big deal, but I like the consistency. It should be a pretty easy thing to eslint and autofix, if others agree.

File: src/components/Responsive/Hidden/Hidden.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React, { useCallback, ReactElement } from 'react';
+import intersection from 'lodash/intersection';
+
+import Context from '../Context';
+
+interface HiddenProps {
+  breakpoints: string[];
```
Comment (pixelbandito) [inline src/components/Responsive/Hidden/Hidden.tsx:7]: Breakpoints wasn't previously required. the lodash `intersection` method is null / undefined safe, and just returns an empty array.

Small, but I think this merits a fix.

File: src/components/Responsive/Hidden/Hidden.test.tsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 import { mount } from 'enzyme';
 import Context from '../Context';
-import Hidden from './';
+import Hidden from '.';
 
 describe('<Hidden />', () => {
   it('renders without error', () => {
     mount(
-      <Hidden>
+      <Hidden breakpoints={[]}>
```
Comment (pixelbandito) [inline src/components/Responsive/Hidden/Hidden.test.tsx:9]: See my comment below - this shouldn't need to be required.
I agree it doesn't make much sense to omit it, but in the spirit of robustness "be conservative in what you do, be liberal in what you accept from others"

Comment (pixelbandito) [reply src/components/Responsive/Hidden/Hidden.test.tsx:9]: Postel's law, if you're not familiar. I love how it was written about TCP specs, but it's also good ethics.
https://en.wikipedia.org/wiki/Robustness_principle

File: src/components/Responsive/Visible/Visible.test.tsx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import Visible from '.';
 describe('<Visible />', () => {
   it('renders without error', () => {
     mount(
-      <Visible>
+      <Visible breakpoints={[]}>
```
Comment (pixelbandito) [inline src/components/Responsive/Visible/Visible.test.tsx:9]: See my comment on `Hidden` - see if we can keep `breakpoints` optional?

File: src/components/Responsive/story.tsx [hunk 6]
```diff
@@ -6,12 +6,13 @@ import { object } from '@storybook/addon-knobs';
...
 import ResponsiveElementReadme from './Element/README.md';
-import Responsive from './';
+import Responsive from '.';
+import { defaultBreakpoints } from './Container/Container';
 
 const MyComponent = ({ responsiveContext }) => (
   <div>
     My component&apos;s `responsiveContext` prop:
-    <pre style={{ whiteSpace: 'wrap', wordWrap: 'break-word' }}>
+    <pre className="break-words">
```
Comment (pixelbandito) [inline src/components/Responsive/story.tsx:15]: 😉

File: src/components/Responsive/Element/Element.tsx (hunk 1 above)
Comment (mdespuits) [reply src/components/Responsive/Element/Element.tsx:4]: No, `React.FC` handling of the `children` prop is so simple and nice. No complaints at all.

**Pull request merged**